app/demo-geant-integration/HitRootIO.cc

```diff
@@ -170,13 +172,17 @@ void HitRootIO::Close()
  */
 void HitRootIO::Merge()
 {
+#if G4VERSION_NUMBER >= 1070
     auto nthreads = G4RunManager::GetRunManager()->GetNumberOfThreads();
+#else
+    auto nthreads = 1;
+#endif
```

Comment:
We'll want to try to reuse the `GetNumberOfThreads` function here so it's not incorrectly 1 for MT cases

Reply:
Doing a similar dynamic cast as for the `Logger` setup and the fancy const variable initialization